### ISSUE TITLE
Validation results stay unchanged after first iteration, possible issue with parameter synchronization in rollout?

### ISSUE THREAD

**Issue by chenweize1998** [volcengine/verl#767]
Hi, I've encountered an issue where the validation results before and after the first iteration of training remain exactly the same, but they change significantly at the second iteration. I suspect that the actor's parameters may not be properly updated in the rollout process immediately after the first optimization step.

Steps to reproduce:
1. Set `trainer.test_freq=1`
2. Set `actor_rollout_ref.actor.optim.lr` to a unreasonably large value (e.g., `0.1`) to ensure that parameter updates have a noticeable effect.
3. Observe that validation results before and after the first iteration are the same, and greatly changes after the second iteration.

Is there a specific place in the code where I should manually trigger a parameter update for the rollout? I think `FSDPVLLMShardingManager` should already handle the parameter synchronization in `__enter__`:

https://github.com/volcengine/verl/blob/8ee3faacb69de40c9fadf486a7e28ef65ed1ba21/verl/workers/sharding_manager/fsdp_vllm.py#L91-L99

but it seems not work as expected? Any pointers to the relevant code section would be helpful.

- Framework version: commit 44a65f956d72aa1bd88ad3ea24de33a309bb0c70 with some minor changes in reward function, possibly unrelated to the issue

**Comment by chenweize1998** [comment]
I think I’ve found the bug. It’s not due to parameter synchronization but rather the learning rate scheduler not being properly updated before the first update. In the current `update_actor` code, the `self.actor_lr_scheduler.step()` is placed *after* the parameter update:
https://github.com/volcengine/verl/blob/591a476dbf5b52028bb6176235664d13dedffddd/verl/workers/fsdp_workers.py#L449-L464
However, before its first `step`, the learning rate remains 0, making the first iteration useless. I think we should move `scheduler.step()` before the parameter update.

**Comment by eric-haibin-lin** [comment]
Yeah good point. Shall we add 1 to current step in the lambda function? Cc @vermouth1992

**Comment by Kirrito-k423** [comment]
定时清理检查结果：这个 issue 最近一次由当前运行账号以外账号产生的公开回复或公开引用是 `2025-03-27T19:54:38Z`。本轮检查日期为 2026-07-04，该时间已经超过 6 个月。检查来源包括 issue comments API `https://api.github.com/repos/verl-project/verl/issues/767/comments?per_page=100` 和 timeline API `https://api.github.com/repos/verl-project/verl/issues/767/timeline?per_page=100`。

按照本仓库的定时 issue 清理标准，我会先留下这条说明并关闭该 issue。若作者或社区成员仍能提供新的复现、证据或需求背景，请继续评论请求重开，或创建新的 issue 并引用这里的上下文。

---

Scheduled cleanup check: the latest public reply or public reference from an account other than the current automation account was `2025-03-27T19:54:38Z`. The check date for this run is 2026-07-04, so that activity is more than 6 months old. The checked public sources were the issue comments API `https://api.github.com/repos/verl-project/verl/issues/767/comments?per_page=100` and the timeline API `https://api.github.com/repos/verl-project/verl/issues/767/timeline?per_page=100`.

Following the scheduled issue cleanup policy, I am leaving this note first and then closing the issue. If the author or community can still provide a fresh reproduction, evidence, or requirement context, please comment to request reopening or open a new issue that links back to this one.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
